### DISCUSSION TITLE
使用redir-host做透明代理，DNS请求是否会多次的疑问，请大佬帮忙解答下

### DISCUSSION THREAD

**Discussion by GoFightNow** [vernesong/OpenClash#4155] [Q&A]
我认为在redir-host的透明代理中，PC发起DNS请求并由国外服务器解析正常IP，那PC拿到这个IP，应该是直接使用IP来发送数据包而不是域名。那这时候Clash收到这个包，我看还是根据域名匹配发起的请求，为什么还是根据域名匹配发起呢？那这个包发送到代理服务器应该还需要解析一次DNS，才能访问网页数据，这样是不是重复DNS请求了呢？如果是，除了fake-ip有什么优化方法吗？

**Comment by fejeIaqbhbv** [comment]
请求dns：我想知道ABC在哪
返回dns：ABC在XYZ
发送数据：帮我把这封信寄给位于XYZ的ABC

**Reply by GoFightNow** [reply]
那最终代理服务器还会发起dns吗？还是说具体情况要看有没有开嗅探

**Reply by fejeIaqbhbv** [reply]
看查询dns的服务器和代理流量服务器是不是一个了
fakeip也一样

**Reply by GoFightNow** [reply]
那保持[respect-rules](https://wiki.metacubex.one/config/dns/#respect-rules)是开启，是不是就绝对是代理服务器发起dns请求，也就是不会再浪费DNS请求。

**Reply by fejeIaqbhbv** [reply]
不绝对，特别是机场用户
代理节点可能由于压力大而清除缓存（也可能干脆就不开缓存）
而且有些机场会用负载均衡，即一个入口对应多个同机房的出口

**Comment by GoFightNow** [comment]
那redir-host基本DNS会重复发起。那fakeip好像不会出现这样的问题吧？本地的dns直接被返回假ip，建立连接请求直接代理服务器dns+网页数据一起就完事儿。

**Reply by fejeIaqbhbv** [reply]
> 本地的dns直接被返回假ip

返回假ip的同时也会去索要真ip

> 建立连接请求直接代理服务器dns+网页数据一起就完事儿。

这就又返回这句话了：

> > 看查询dns的服务器和代理流量服务器是不是一个了

**Reply by GoFightNow** [reply]
那就是说fake-ip也会先请求一遍DNS然后在发给服务器，然后有几率在请求一次嘛。那fakeip有点太没用了。。
